### ISSUE TITLE
Tool Tips reporting erronious information

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#35419]
VSF_TYPE_MARKDOWNTool Tips is reporting that a type variable with the same name as a class is the class.

Example code:
```
void Main()
{
	var thing1 = new MyFoo();
	var thing2 = new Record();
	var thing3 = new Bar();
	Console.WriteLine(thing1.Method(thing2));
	Console.WriteLine(thing1.Method(thing3));
}

public abstract class Foo
{
	public abstract int Method&lt;T&gt;(T input) where T : IFoo;
}

public interface IFoo
{
	bool Set { get; }
}

public class Bar : IFoo
{
	public bool Set =&gt; false;
}

public class Record : IFoo
{
	public bool Set =&gt; true;
}

public class MyFoo : Foo
{
	public override int Method&lt;Record&gt;(Record input)
	{		
		if (input.Set)
			return 1;
			
		return 0;
	}
}
```

Tool tips is showing that for this line: `public override int Method&lt;Record&gt;(Record input)` 'Record' is the class Record and not a type variable called 'Record', which is confusing. If things were as the tool tips is reporting than this line: `Console.WriteLine(thing1.Method(thing3));` should throw a compiler error. But the program runs just fine.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/525287/tool-tips-reporting-erronious-information.html
VSTS ticketId: 843615_
_These are the original issue comments:_

Visual Studio Feedback System on 4/10/2019, 00:49 AM (21 days ago): <p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

_These are the original issue solutions:_
(no solutions)

**Comment by CyrusNajmabadi** [comment]
I don't repro this at all.  The tooltip is correct:

![image](https://user-images.githubusercontent.com/4564579/57055198-a7cd2880-6c4e-11e9-8acd-c720a4f54a46.png)

**Comment by gafter** [comment]
I was unable to repro any issue with the tooltips.
